pkg/cloud/aws/services/ec2/instances.go

```diff
@@ -138,41 +139,55 @@ func (s *Service) createInstance(machine *actuators.MachineScope, bootstrapToken
 		)
 	}
 
-	// apply values based on the role of the machine
-	if machine.Role() == "controlplane" {
+	// capture current control plane status
+	controlPlaneNodeCount := machine.ClusterControlPlaneStatus().ControlPlaneNodeCount
 
+	// apply values based on the role of the machine
+	switch machine.Role() {
+	case "controlplane":
 		if s.scope.SecurityGroups()[v1alpha1.SecurityGroupControlPlane] == nil {
 			return nil, awserrors.NewFailedDependency(
 				errors.New("failed to run controlplane, security group not available"),
 			)
 		}
 
-		if len(s.scope.ClusterConfig.CAPrivateKey) == 0 {
-			return nil, awserrors.NewFailedDependency(
-				errors.New("failed to run controlplane, missing CAPrivateKey"),
-			)
-		}
-
-		userData, err := userdata.NewControlPlane(&userdata.ControlPlaneInput{
-			CACert:            string(s.scope.ClusterConfig.CACertificate),
-			CAKey:             string(s.scope.ClusterConfig.CAPrivateKey),
-			ELBAddress:        s.scope.Network().APIServerELB.DNSName,
-			ClusterName:       s.scope.Name(),
-			PodSubnet:         s.scope.Cluster.Spec.ClusterNetwork.Pods.CIDRBlocks[0],
-			ServiceSubnet:     s.scope.Cluster.Spec.ClusterNetwork.Services.CIDRBlocks[0],
-			ServiceDomain:     s.scope.Cluster.Spec.ClusterNetwork.ServiceDomain,
-			KubernetesVersion: machine.Machine.Spec.Versions.ControlPlane,
-		})
+		var userData string
+		var err error
+
+		if controlPlaneNodeCount > 1 {
+			// TODO: WIP controlplane node join:
+			// get kubeadm node join token
+			// prepare userdata with kubeadm join https://kubernetes.io/docs/setup/independent/high-availability/
+
+			klog.V(2).Infof("Allowing machine %q to join control plane for cluster %q", machine.Name(), s.scope.Name())
+		} else {
+			klog.V(2).Infof("Machine %q is the first control plane node for cluster %q", machine.Name(), s.scope.Name())
+			if len(s.scope.ClusterConfig.CAPrivateKey) == 0 {
+				return nil, awserrors.NewFailedDependency(
+					errors.New("failed to run controlplane, missing CAPrivateKey"),
+				)
+			}
+
+			userData, err = userdata.NewControlPlane(&userdata.ControlPlaneInput{
+				CACert:            string(s.scope.ClusterConfig.CACertificate),
+				CAKey:             string(s.scope.ClusterConfig.CAPrivateKey),
+				ELBAddress:        s.scope.Network().APIServerELB.DNSName,
+				ClusterName:       s.scope.Name(),
+				PodSubnet:         s.scope.Cluster.Spec.ClusterNetwork.Pods.CIDRBlocks[0],
+				ServiceSubnet:     s.scope.Cluster.Spec.ClusterNetwork.Services.CIDRBlocks[0],
+				ServiceDomain:     s.scope.Cluster.Spec.ClusterNetwork.ServiceDomain,
+				KubernetesVersion: machine.Machine.Spec.Versions.ControlPlane,
+			})
 
-		if err != nil {
-			return input, err
+			if err != nil {
+				return input, err
+			}
 		}
+		controlPlaneNodeCount++
```

Comment:
Should we make these a function on the ControlPlane status instead of modifying the value?

Reply:
Dropping this approach